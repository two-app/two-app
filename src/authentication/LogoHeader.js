// @flow
import React from "react";
import {StyleSheet, Text} from "react-native";
import PropTypes from "prop-types";
import Colors from "../Colors";

const LogoHeader = ({heading}) => (
    <>
        <Text style={styles.logo}>two.</Text>
        <Text style={styles.heading} id="heading">{heading}</Text>
    </>
);

LogoHeader.propTypes = {
    heading: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
    logo: {
        color: Colors.SALMON,
        fontSize: 50,
        fontWeight: 'bold',
        textAlign: 'center',
        paddingTop: 20,
        paddingBottom: 20
    },
    heading: {
        fontSize: 25,
        fontWeight: "bold",
        color: Colors.DARK
    }
});

export default LogoHeader;