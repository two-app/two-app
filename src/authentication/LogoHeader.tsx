import React from 'react';
import {StyleSheet, Text} from 'react-native';
import Colors from '../Colors';

const LogoHeader = ({heading}: { heading: string }) => (
    <>
        <Text style={styles.logo}>two.</Text>
        <Text style={styles.heading} id="heading">{heading}</Text>
    </>
);

const styles = StyleSheet.create({
    logo: {
        color: Colors.SALMON,
        fontSize: 50,
        fontWeight: 'bold',
        textAlign: 'center',
        paddingTop: 20,
        paddingBottom: 20
    },
    heading: {
        fontSize: 25,
        fontWeight: 'bold',
        color: Colors.DARK
    }
});

export default LogoHeader;