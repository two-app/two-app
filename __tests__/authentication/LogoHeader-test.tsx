import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import LogoHeader from '../../src/authentication/LogoHeader';

test('should maintain snapshot', () => expect(
    renderer.create(<LogoHeader heading="Test Heading"/>)
).toMatchSnapshot());

test('should display given heading', () => expect(
    shallow(<LogoHeader heading="Test Heading"/>).find('Text[data-testid=\'heading\']').render().text()
).toEqual('Test Heading'));