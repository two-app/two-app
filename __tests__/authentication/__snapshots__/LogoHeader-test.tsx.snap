// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should maintain snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#EE7674",
        "fontSize": 50,
        "fontWeight": "bold",
        "paddingBottom": 20,
        "paddingTop": 20,
        "textAlign": "center",
      }
    }
  >
    two.
  </Text>,
  <Text
    data-testid="heading"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.8)",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    Test Heading
  </Text>,
]
`;
